ibm/mas_devops/roles/amqstreams/tasks/main.yaml

```diff
@@ -69,24 +69,21 @@
 # We have seen instances where the Kafka cluster is reported as Ready but
 # the necessary fields in the status resource are not set yet so we retry until
 # the expected fields are set as well
-- name: "Lookup Kafka Resource"
+- name: "Wait for Kafka Resource to be ready (60s delay)"
   community.kubernetes.k8s_info:
     api_version: kafka.strimzi.io/v1beta2
     kind: Kafka
     name: "{{ kafka_cluster_name }}"
     namespace: "{{ kafka_namespace }}"
-    wait: yes
-    wait_sleep: 10
-    wait_timeout: 300 # 5 mins until we give up waiting for the cluster to be ready
-    wait_condition:
-      type: Ready
-      status: "True"
   register: kafka_cluster_lookup
-  retries: 120 # Once the cluster is ready we will wait for up to 20 minutes for the listener information to be added to the status sub-resource
-  delay: 10 # seconds
+  retries: 30 # will wait for up to 30 minutes for the listener information to be added to the status sub-resource
+  delay: 60 # seconds
   until:
+    - kafka_cluster_lookup.resources is defined
     - kafka_cluster_lookup.resources | length > 0
     - kafka_cluster_lookup.resources[0].status is defined
+    - kafka_cluster_lookup.resources[0].status.conditions[0].type == "Ready"
```

Comment:
This seems risky ... the first condition might not always be the ready condition.  We should use code something like this to check for presence of a Ready condition anywhere in the status, and then confirm the status is set to "True":

```
- kafka_cluster_lookup.resources[0].status.conditions | selectattr('type', 'equalto', 'Ready') | map(attribute='status') | list | length > 0
- kafka_cluster_lookup.resources[0].status.conditions | selectattr('type', 'equalto', 'Ready')| map(attribute='status') | list | first == "True"
```